### DISCUSSION TITLE
Quickly access assets with site.com/{assetTag}

### DISCUSSION THREAD

**Discussion by MrCaspan** [snipe/snipe-it#9301] [Q&A] [answered]
It would be really nice to have the ability to quickly access assets using site.com/{assetTag}

Not asset ID but TAG this would make it easy for an admin with a mobile phone to quickly access the asset and check it out or check it in.

Thanks for the awesome app again!!

**Comment by MrCaspan** [comment]
After thinking about this it would be nice to also have a programable entry point to the inventory system using the asset tag number. Ill give an example:

We have QR code asset tags that were made up a long time ago. When you scan the QR code it points to https://inventory.site.com/id/{assetTag} Now that we have a new inventory system I was able to add to the .htaccess file a mod_rewrite that would redirect any requests that match 'id/[0-9] as our asset tags are only numbers and then I took the asset tag number and redirected the user to https://inventory.site.com/hardware/bytag?assetTag={assetTag} 

So I was jut thinking I'm sure others might have this issue when coming from another system where they had unique URLs that pointed to their assets like this. Might be cool to put into the system a config page so the system handles this request and not the .htaccess. On this page you could tell it the URL to expect and what to strip of what part is the asset tag and do the redirect.

If you combine that with the first option you can just redirect them to https://inventory.site.com/{assetTag} Just more of my constant thinking to improve stuff sorry, I'm just worried the .htaccess will be overridden on the next update!

**Comment by snipe** [comment, marked as answer]
We already have that functionality.

`https://snipe-it.local:8890/hardware/bytag?assetTag=XXXXXXX`

It's how the top search bar works.

**Reply by MrCaspan** [reply]
My only suggestion was more to make a URL that is programmable you can set in the system settings so if a user has existing asset tags like we have and spent a lot of money on them that point to http://domain.com/i/{asset tag} with QR codes on them. I have had to create a capture for it in the .htaccess (That will get over written each update) If it was programmable you can make old asset tags that have QR codes on them still work with the Snipe-IT system. Just makes this app much more diverse and usable coming from another system.Just an idea. I have a workaround for now to get my old asset tags to work with your system using htaccess rewire. Just trying to make the system better :)

**Reply by snipe** [reply]
Do you mean have an *arbitrary* pattern that would match some old tags? Like if you were using asset panda and your user was `assets.foo.com/some-arbitrary-directory/asset_tag` and you wanted to use us instead, and you didn't want to reprint labels?

I think .htaccess would still be the best solution here to be honest. The kind of routing overhead we'd have to support - and worse yet, the kind of weird heisenbugs we'd run into when adding new features that conflict with those urls would be a nightmare.

For example, we already see this with the /liceses` directory on XAMPP. It's a "magical" directory for that web server, where the Apache server's license is kept, but if you don't override it manually using an apache rule, you can't see the Snipe-IT licenses directory.  https://snipe-it.readme.io/docs/xampp#remove-built-in-aliases

So let's say your arbitrary, user-defined URL is `assets.foo.com/some-arbitrary-directory/asset_tag`, and we end up creating a feature whose url is  `assets.foo.com/some-arbitrary-directory`, it would be a nightmare to debug.

To be clear, I think it's a cool idea - I just don't see any way to make it work without being a horrible nightmare.

**Reply by MrCaspan** [reply]
haha completely understand! It was just a thought but you give great reasons!

**Comment by snipe** [comment]
Unfortunately, I don’t see a way to mark this discussion as “answered”. (I think discussions are still kind of a beta feature for GH.)

**Reply by snipe** [reply]
Ohhh, I bet you can’t mark things in the “ideas” category as “answered”.
